test/Spago/Build.purs

```diff
@@ -99,65 +101,212 @@ spec = Spec.around withTempDir do
+            { package = config.package <#> \r -> r
+                { test = Just
+                    { main: "Test.Main"
+                    , pedantic_packages: Just true
+                    , strict: Nothing
+                    , censor_test_warnings: Nothing
+                    , dependencies: maybe (Config.Dependencies Map.empty) _.dependencies r.test
+                    , execArgs: r.test >>= _.execArgs
+                    }
+                }
+            }
 
-      Spec.describe "fails when there are imports from transitive dependencies and" do
-        let
-          setupSrcTransitiveTests spago = do
-            spago [ "init", "--name", "7368613235362d34312f4e59746b7869335477336d33414d72" ] >>= shouldBeSuccess
-            spago [ "install", "maybe" ] >>= shouldBeSuccess
-            FS.writeTextFile (Path.concat [ "src", "Main.purs" ]) "module Main where\nimport Prelude\nimport Data.Maybe\nimport Control.Alt\nmain = unit"
-            -- get rid of "Compiling ..." messages and other compiler warnings
-            spago [ "build" ] >>= shouldBeSuccess
+      Spec.describe "fails when imports from transitive dependencies" do
 
-        Spec.it "passed --pedantic-packages CLI flag" \{ spago, fixture } -> do
-          setupSrcTransitiveTests spago
-          spago [ "build", "--pedantic-packages" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency.txt")
+        Spec.describe "appear in the source package and" do
 
-        Spec.it "package config has 'pedantic_packages: true'" \{ spago, fixture } -> do
-          setupSrcTransitiveTests spago
-          addPedanticPackagesToSrc
-          spago [ "build" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency.txt")
+          let
+            setupSrcTransitiveTests spago installConsoleAndEffect = do
+              spago [ "init", "--name", "7368613235362d34312f4e59746b7869335477336d33414d72" ] >>= shouldBeSuccess
+              spago [ "install", "maybe" ] >>= shouldBeSuccess
+              when installConsoleAndEffect do
+                spago [ "install", "--test-deps", "console", "effect" ] >>= shouldBeSuccess
+              FS.writeTextFile (Path.concat [ "src", "Main.purs" ]) $ Array.intercalate "\n"
+                [ "module Main where"
+                , "import Prelude"
+                , "import Data.Maybe as Maybe"
+                , "import Effect as Effect"
+                , "import Effect.Console as Console"
+                , "import Control.Alt as Alt"
+                , "main = unit"
+                ]
+              -- get rid of "Compiling ..." messages and other compiler warnings
+              spago [ "build" ] >>= shouldBeSuccess
+
+          Spec.it "passed --pedantic-packages CLI flag" \{ spago, fixture } -> do
+            setupSrcTransitiveTests spago true
+            spago [ "build", "--pedantic-packages" ] >>= shouldBeFailureErr (fixture "check-direct-import-transitive-dependency.txt")
```

Comment:
I'd expect this to complain about the test packages containing superfluous `console` and `effect` (since they are already in the source dependencies) - the flag should turn it on for both src and test targets, since it's a one-off, i.e. the user is explicitly asking for a cleanup pass

Reply:
It did. That's why the `true` arg in `setupSrcTransitiveTests spago true` installs those deps in `test`. Otherwise, I'll get an error about that. In this particular group of tests, we're testing whether source packages alone get that error, which I can't do since `--pedantic-packages` enables that check on the `test` code as well.

Reply:
I think my confusion comes from the fact that it's hard to see what the test is trying to verify - so I suggest we:
- do not generate the outputs via code, but only use fixtures
- this means that we'll have a lot of fixtures, so let's stick them in a separate folder, like `test-fixtures/pedantic`
- then before each test case let's have a few lines summarising why the test should return what it returns

Reply:
That's... a lot of work. Is there an alternative for making these tests clearer such as comments?

Reply:
We can merge and I can do the work if you'd prefer

Reply:
I can also pile commits on this branch if you'd like a self-contained package

Reply:
I'm fine with you pushing commits here.